portfolio/src/main/java/com/google/sps/servlets/CommentServlet.java

```diff
@@ -0,0 +1,110 @@
+package com.google.sps.servlets;
+
+import com.google.sps.classes.Comment;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.gson.Gson;
+import java.io.IOException;
+import java.util.Date;
+import java.util.ArrayList;
+import java.util.List;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/add-comment")
+public class CommentServlet extends HttpServlet {
+  
+ @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+
+    int entityCount=-1;
+
+    Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+
+    List<Comment> commentList = new ArrayList<>();
+
+    for (Entity entity: results.asIterable()) {
+
+      String name = (String) entity.getProperty("name");
+      String message = (String) entity.getProperty("message");
+      Date time = (Date) entity.getProperty("timestamp");
+
+      commentList.add(new Comment(name, message, time));
+      entityCount++;
+    }
+
+    int numComments = getNumComments(request);
+
+    if (numComments>entityCount){
+        numComments=entityCount;
+    }
+
+    commentList = commentList.subList(0, numComments);
+
+    String json = convertToJsonUsingGson(commentList);
+
+    response.setContentType("application/json;");
+    response.getWriter().println(json);
+  }
+
+  private int getNumComments(HttpServletRequest request){
+    String userChoiceString = request.getParameter("commentsQuantity");
+
+    int userChoice;
+    try {
+      userChoice = Integer.parseInt(userChoiceString);
+    } 
+    catch (NumberFormatException e) {
+      System.err.println("Could not convert to int: " + userChoiceString);
+      return -1;
```

Comment:
don't be afraid to throw exceptions, especially when the special case int is not really helping much - I think you will get an exception in the sublist call.

In general, we try to follow https://en.wikipedia.org/wiki/Fail-fast philosophy.